File: ScaleformUI_Csharp/Menus/UIMenu/SidePanels/DetailsPanel/UIFreemodeDetailsItem.cs
using ScaleformUI.Elements;
using ScaleformUI.PauseMenu;
using ScaleformUI.PauseMenus.Elements;

namespace ScaleformUI.Menu
{
    public class UIFreemodeDetailsItem : PauseMenuItem
    {
        public string TextRight = "";
        public BadgeIcon Icon = BadgeIcon.NONE;
        public SColor IconColor = SColor.HUD_None;
        public int Type;
        public bool Tick;
        internal ItemFont _rightLabelFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY;
        internal CrewTag CrewTag = new CrewTag();

        public UIFreemodeDetailsItem(string description): this(description, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY)
        {
        }
        public UIFreemodeDetailsItem(string description, ItemFont labelFont) : base(description, labelFont)
        {
            Type = 5;
            LabelFont = labelFont;
        }
        public UIFreemodeDetailsItem(string textLeft, string textRight, bool separator) : this(textLeft, textRight, separator, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY)
        {
        }

        public UIFreemodeDetailsItem(string textLeft, string textRight, bool separator, ItemFont labelFont) : this(textLeft, textRight, separator, labelFont, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY)
        {
        }

        public UIFreemodeDetailsItem(string textLeft, string textRight, bool separator, ItemFont labelFont, ItemFont rightLabelFont) : base(textLeft, labelFont)
        {
            Type = separator ? 4 : 0;
            TextRight = textRight;
            _rightLabelFont = rightLabelFont;
        }

        public UIFreemodeDetailsItem(string textLeft, CrewTag crewTag) : this(textLeft, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY, crewTag)
        {
        }

        public UIFreemodeDetailsItem(string textLeft, ItemFont labelFont, CrewTag crewTag) : base(textLeft, labelFont)
        {
            Type = 3;
            CrewTag = crewTag;
        }

        public UIFreemodeDetailsItem(string textLeft, string textRight, BadgeIcon icon, SColor iconColor, bool tick = false) : this(textLeft, textRight, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY, icon, iconColor, tick)
        {
        }
 
        public UIFreemodeDetailsItem(string textLeft, string textRight, ItemFont labelFont, BadgeIcon icon, SColor iconColor, bool tick = false) : this(textLeft, textRight, labelFont, ScaleformFonts.CHALET_LONDON_NINETEENSIXTY, icon, iconColor, tick)
        {
        }

        public UIFreemodeDetailsItem(string textLeft, string textRight, ItemFont labelFont, ItemFont rightLabelFont, BadgeIcon icon, SColor iconColor, bool tick = false) : base(textLeft, labelFont)
        {
            Type = 2;
            TextRight = textRight;
            Icon = icon;
            IconColor = iconColor;
            Tick = tick;
            _rightLabelFont = rightLabelFont;
        }
    }
}

File: ScaleformUI_V8/src/menus/UIMenu/sidepanels/DetailsPanel/uifreemodedetailsitem.ts
import { BadgeStyle } from "elements/badge";
import { ItemFont } from "elements/item-font";
import { ScaleformFonts } from "elements/scaleform-fonts";
import { SColor } from "elements/scolor";

export class UIFreemodeDetailsItem {
    public TextLeft: string;
    public TextRight: string = "";
    public Icon: BadgeStyle = BadgeStyle.NONE;
    public IconColor: SColor = SColor.HUD_None;
    public Type: number;
    public Tick: boolean;
    _labelFont: ItemFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY;
    _rightLabelFont: ItemFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY;

    constructor(textLeft: string, textRight: string, seperator: boolean, icon: BadgeStyle, iconColor: SColor, tick: boolean) {
        let _type: number;
        if (seperator)
            _type = 3
        else if (icon != undefined && iconColor != undefined)
            _type = 2
        else if (textRight == undefined && seperator == undefined && icon == undefined && iconColor == undefined && tick == undefined)
            _type = 4
        else
            _type = 0

        this.Type = _type;
        this.TextLeft = textLeft;
        this.TextRight = textRight || "";
        this.Icon = icon || BadgeStyle.NONE;
        this.IconColor = iconColor || SColor.HUD_White;
        this.Tick = tick || false;
        this._labelFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY;
        this._rightLabelFont = ScaleformFonts.CHALET_LONDON_NINETEENSIXTY
    }

    SetLabelsFonts(leftFont: ItemFont, rightFont: ItemFont) {
        if (leftFont != undefined)
            this._labelFont = leftFont
        if (rightFont != undefined)
            this._rightLabelFont = rightFont
    }

}